workflow_array_ephys/ingest.py

```diff
@@ -1,90 +1,127 @@
                 probe_number = int(probe_number.replace('imec', ''))
 
-                insertions.append({'probe': spikeglx_meta.probe_SN, 'insertion_number': int(probe_number)})
+                insertions.append({'probe': spikeglx_meta.probe_SN,
+                                   'insertion_number': int(probe_number)})
                 session_datetimes.append(spikeglx_meta.recording_time)
         elif acq_software == 'OpenEphys':
             loaded_oe = openephys.OpenEphys(session_dir)
             session_datetimes.append(loaded_oe.experiment.datetime)
             for probe_idx, oe_probe in enumerate(loaded_oe.probes.values()):
-                probe_key = {'probe_type': oe_probe.probe_model, 'probe': oe_probe.probe_SN}
-                if probe_key['probe'] not in [p['probe'] for p in probe_list] and probe_key not in probe.Probe():
+                probe_key = {'probe_type': oe_probe.probe_model,
+                             'probe': oe_probe.probe_SN}
+                if (probe_key['probe'] not in [p['probe'] for p in probe_list
+                                               ] and probe_key not in probe.Probe()):
                     probe_list.append(probe_key)
-                insertions.append({'probe': oe_probe.probe_SN, 'insertion_number': probe_idx})
+                insertions.append({'probe': oe_probe.probe_SN,
+                                   'insertion_number': probe_idx})
         else:
-            raise NotImplementedError(f'Unknown acquisition software: {acq_software}')
+            raise NotImplementedError('Unknown acquisition software: '
+                                      + f'{acq_software}')
 
         # new session/probe-insertion
-        session_key = {'subject': sess['subject'], 'session_datetime': min(session_datetimes)}
+        session_key = {'subject': sess['subject'],
+                       'session_datetime': min(session_datetimes)}
         if session_key not in session.Session():
             session_list.append(session_key)
-            session_dir_list.append({**session_key, 'session_dir': session_dir.relative_to(root_data_dir).as_posix()})
-            probe_insertion_list.extend([{**session_key, **insertion} for insertion in insertions])
-
-    print(f'\n---- Insert {len(session_list)} entry(s) into session.Session ----')
+            root_dir = find_root_directory(get_ephys_root_data_dir(), session_dir)
+            session_dir_list.append({**session_key, 'session_dir':
+                                     session_dir.relative_to(root_dir).as_posix()})
+            probe_insertion_list.extend([{**session_key, **insertion
+                                          } for insertion in insertions])
+
+    if verbose:
+        previous_length = len(session.Session.fetch())
     session.Session.insert(session_list)
     session.SessionDirectory.insert(session_dir_list)
+    if verbose:
+        insert_length = len(session.Session.fetch()) - previous_length
```

Comment:
Perhaps we don't need this logic because of the above statement: `if session_key not in session.Session()`?

Reply:
Probably the same is the case for the tables below (`probe.Probe`, `ephys.ProbeInsertion`).

Reply:
You're right - I over-applied this logic. Good catch, thanks